src/string.cr

```diff
@@ -2405,25 +2404,52 @@ class String
   def compare(other : String, case_insensitive = false)
     return self <=> other unless case_insensitive
 
-    reader1 = Char::Reader.new(self)
-    reader2 = Char::Reader.new(other)
-    ch1 = reader1.current_char
-    ch2 = reader2.current_char
-
-    while reader1.has_next? && reader2.has_next?
-      cmp = ch1.downcase <=> ch2.downcase
-      return cmp.sign if cmp != 0
+    if ascii_only? && other.ascii_only?
+      if self.size > other.size
+        return 1
+      elsif self.size < other.size
+        return -1
+      end
```

Comment:
These greater-or-less checks are pretty ugly and generally make this method slightly slower I think.
What is the usecase for a method which in addition to being case-insensitive also returns which of the two is greater or smaller? I think this feature is/will be very rarely used and mostly a ` == 0` will be put after this method.
I really wish this method would just return `true` or `false` and would be case-insensitive by default...

Reply:
The method is named `#compare` not `#equals?`.

This boils down to 2 integer comparisons, performance impact is likely negligible compared to the whole method.

Reply:
Yes that was basically a request to rename it to something like `equals?` and make it case-insensitive by default.

Actually 4 integer comparisons. One before the loop (these two^) and 2 integer comparisons in every single iteration of `@bytesize.times do |i|`. I would really like to avoid these and make the `== 0` redundant.

Reply:
Thanks with that it's as fast as [this](https://gist.github.com/r00ster91/ef1cdfd0c20b3ab2f69918c43a8e9e83#file-string-cr) now.
Any opinions on making it this instead of the current `compare` though?

Reply:
If you write the method that compares case-insensitive and returns `true` or `false`, how it performs compared to this?

Reply:
(but in any case sometimes knowing less-than or greater-than is good, so we won't change the fact that it returns a number)

Reply:
Hey, I think this condition is wrong: if a string is longer/shorter than other it doesn't mean it's lower/greater. For example "z" is greater than "hello", while "h" is lower than "zzz".